server/modules/arenaImport/service/arenaImport/arenaImportJob.js

```diff
@@ -70,16 +68,28 @@ export default class ArenaImportJob extends Job {
   async onEnd() {
     await super.onEnd()
 
-    const { arenaSurveyFileZip, surveyId, filePath } = this.context
+    const { arenaSurveyFileZip, backup, filePath, surveyId } = this.context
 
     if (arenaSurveyFileZip) {
       arenaSurveyFileZip.close()
     }
 
-    if (!this.isSucceeded() && surveyId) {
-      await SurveyManager.dropSurveySchema(surveyId)
+    if (surveyId) {
+      if (backup) {
+        if (this.isSucceeded()) {
+          this.logDebug(`removing 'temporary' flag from survey ${surveyId}...`)
+          await SurveyManager.removeSurveyTemporaryFlag({ surveyId })
+          this.logDebug(`'temporary' flag removed from survey ${surveyId}`)
+        } else {
+          this.logDebug(`deleting temporary survey ${surveyId}...`)
+          await SurveyManager.deleteSurvey(surveyId, { deleteUserPrefs: false })
+          this.logDebug(`survey ${surveyId} deleted!`)
+        }
```

Comment:
this part is common to 2 or 3 jobs... it's nice to show logs for it; where can we put it? in some kind of SurveyCreatorJobHelper module?!